sound/soc/sof/trace.c

```diff
@@ -13,6 +13,214 @@
 #include "sof-priv.h"
 #include "ops.h"
 
+#define TRACE_FILTER_ELEMENTS_PER_ENTRY 4
+
+static int trace_filter_append_elem(int32_t key, int32_t value,
+				    struct sof_ipc_trace_filter_elem *elem_list,
+				    int capacity, int *counter)
+{
+	if (*counter >= capacity)
+		return -ENOMEM;
+
+	elem_list[*counter].key = key;
+	elem_list[*counter].value = value;
+	++*counter;
+
+	return 0;
+}
+
+static int trace_filter_parse_entry(struct snd_sof_dev *sdev, const char *line,
+				    struct sof_ipc_trace_filter_elem *elem,
+				    int capacity, int *counter)
+{
+	int len = strlen(line);
+	int cnt = *counter;
+	int log_level;
+	int uuid_id;
+	int pipe_id;
+	int comp_id;
+	int read;
+	int ret;
+
+	/* ignore empty content */
+	if(!line[0])
+		return 0;
+
+	ret = sscanf(line, " %d %x %d %d %n", &log_level, &uuid_id, &pipe_id, &comp_id, &read);
```

Comment:
does the firmware understand -1 as a wildcard for all parameters.

Not sure I get the idea about the 0 uuid? why not all 0b1 as well?

Reply:
sorry, I don't get the answer and the link between uuid and INT32_MAX?

Reply:
>sorry, I don't get [...] link between uuid and INT32_MAX

UUID_id created from memory address, so may have almost any value from 1 to UINT32_MAX range (0 is reserved for NULL), signed value (int32_t) is just to small.

> sorry, I don't get the answer

So I think I didn't understand the question. If there is still some doubts please try to rephrase your question.